style/animation.rs

```diff
@@ -1015,11 +1015,55 @@ impl ElementAnimationSet {
             if transition.state == AnimationState::Finished {
                 continue;
             }
-            if transitioning_properties.contains(transition.property_animation.property_id()) {
+            // Step 3 in https://drafts.csswg.org/css-transitions/#starting
+            // "If the element has a running transition or completed transition for the property, 
+            // and there is not a matching transition-property value, then implementations 
+            // must cancel the running transition or remove the completed transition from the set of completed transitions."
+            if !transitioning_properties.contains(transition.property_animation.property_id()) {
+                transition.state = AnimationState::Canceled;
+                self.dirty = true;
                 continue;
             }
-            transition.state = AnimationState::Canceled;
-            self.dirty = true;
+            
+            let transition_property_id = transition.property_animation.property_id();
+            // Never panic: We made sure in fn start_transition_if_applicable if a transition is valid, after_change_to is also valid
+            let after_change_to = AnimationValue::from_computed_values(transition_property_id, &after_change_style).unwrap();
+            // Step 4
+            // "If the element has a running transition for the property, there is a matching transition-property value, 
+            // and the end value of the running transition is not equal to the value of the property in the after-change style, then:"
+            if transition.property_animation.to != after_change_to {
+                // index for after-change transition declaration
+                let index = after_change_style.transition_properties().position(|declared_transition| declared_transition.property
+                    .as_borrowed()
+                    .to_physical(after_change_style.writing_mode) == transition_property_id).unwrap();
+
+                let now = context.current_time_for_animations;
+                let after_change_style = after_change_style.get_ui();
+                let allow_discrete: bool = after_change_style.transition_behavior_mod(index) == TransitionBehavior::AllowDiscrete;
+                let not_transitionable = !transition_property_id.is_animatable() || (!allow_discrete && transition_property_id.is_discrete_animatable());
```

Comment:
A property like `width` is animatable and not discretely animatable. But still, some values are not transitionable, e.g. from `0px` to `max-content` with `transition-behavior: normal`

See https://github.com/servo/stylo/pull/115